tests/lib/sample_rate_converter/src/main.c

```diff
@@ -0,0 +1,1065 @@
+{
+	int ret;
+
+	uint32_t input_sample_rate = 48000;
+	uint32_t output_sample_rate = 24000;
+	uint32_t conversion_ratio = input_sample_rate / output_sample_rate;
+
+	uint32_t input_samples[] = {1000, 2000, 3000, 4000,  5000,  6000,
+				    7000, 8000, 9000, 10000, 11000, 12000};
+	size_t num_samples = ARRAY_SIZE(input_samples);
+	size_t expected_output_samples = num_samples / conversion_ratio;
+	uint32_t output_samples[expected_output_samples];
+
+	enum sample_rate_converter_filter filter = SAMPLE_RATE_FILTER_SIMPLE;
+	size_t output_size;
+
+	ret = sample_rate_converter_process(
+		&conv_ctx, filter, input_samples, num_samples * sizeof(uint32_t), input_sample_rate,
+		output_samples, expected_output_samples * sizeof(uint16_t), &output_written,
+		output_sample_rate);
+
+	zassert_equal(ret, 0, "Sample rate conversion process failed");
+	zassert_equal(conv_ctx.input_sample_rate, input_sample_rate,
+		      "Input sample rate not as expected");
+	zassert_equal(conv_ctx.output_sample_rate, output_sample_rate,
+		      "Output sample rate not as expected");
+	zassert_equal(conv_ctx.conversion_ratio, conversion_ratio,
+		      "Conversion ratio not as expected");
+	zassert_equal(conv_ctx.conversion_direction, CONVERSION_DIR_DOWN,
+		      "Conversion direction is not as expected");
+	zassert_equal(conv_ctx.filter_type, filter, "Filter set incorrectly");
+	zassert_equal(conv_ctx.input_buf.bytes_in_buf, 0, "Bytes in input buffer not as expected");
+	zassert_equal(ring_buf_size_get(&conv_ctx.output_ringbuf), 0,
+		      "Number of bytes in output ringbuffer not as expected %d",
+		      ring_buf_size_get(&conv_ctx.output_ringbuf));
+
+	zassert_equal(output_written, expected_output_samples * sizeof(uint32_t),
+		      "Output size was not as expected (%d)", output_written);
+
+	/* Verify output bytes, expected two input samples per output sample */
+	for (int i = 0; i < expected_output_samples; i++) {
+		zassert_within(output_samples[i], input_samples[i * 2], 550);
+	}
+}
+
+ZTEST(suite_sample_rate_converter, test_init_valid_decimate_16khz_32bit)
+{
+	int ret;
+
+	uint32_t input_sample_rate = 48000;
+	uint32_t output_sample_rate = 16000;
+	uint32_t conversion_ratio = input_sample_rate / output_sample_rate;
+
+	int32_t input_samples[] = {1000, 2000, 3000, 4000,  5000,  6000,
+				   7000, 8000, 9000, 10000, 11000, 12000};
+	size_t num_samples = ARRAY_SIZE(input_samples);
+	size_t expected_output_samples = num_samples / conversion_ratio;
+	int32_t output_samples[expected_output_samples];
+
+	enum sample_rate_converter_filter filter = SAMPLE_RATE_FILTER_SIMPLE;
+	size_t output_written;
+
+	ret = sample_rate_converter_process(
+		&conv_ctx, filter, input_samples, num_samples * sizeof(uint32_t), input_sample_rate,
+		output_samples, expected_output_samples * sizeof(uint32_t), &output_written,
+		output_sample_rate);
+
+	zassert_equal(ret, 0, "Sample rate conversion process failed");
+	zassert_equal(conv_ctx.input_sample_rate, input_sample_rate,
```

Comment:
Suggest that these checks which are repeated are moved to a static function.

Reply:
I prefer to have all checks in the test, I think that makes it more clear what is tested and easier to understand the test case when debugging.